infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -628,6 +629,11 @@ public Map<LogReplicationSession, ReplicationStatus> queryReplicationStatus() {
             return replicationStatusMap;
         }
 
+        if (sessionManager == null) {
+            log.warn("Log Replication Service has not been bootstrapped yet.");
```

Comment:
How about this warning message: QueryReplicationStatus returns empty as session manager is not started yet?

Reply:
Personally I prefer not to include "session manager" in this log msg, since it's a LR internal concept, and this log is for client usage where they might only be aware of "log replication"

Reply:
Sure, you can avoid session manager. But, the current message (Log Replication Service has not been bootstrapped yet.) does not give any useful information on the operation itself. 

Can someone figure out if QueryReplicationStatus is returning from here?

Reply:
I don't understand what do you mean by "useful information". If you ever check the code path, the fact that this log msg is hit already shows that status query comes in a very small time window where cluster configuration pushed to LR node but the bootstrap yet to complete, which is what the log msg means.

For what "QueryReplicationStatus is returning", check the client code and the entire query response is printed out there. We now have 3 conditions that the returning map will be empty, instead of the fact that the returning map is empty, the reason and code path caused that empty returning is much more useful, which is why we print 3 different log msgs here.